base/native_library_win.cc

```diff
@@ -7,17 +7,152 @@
 #include <windows.h>
 
 #include "base/files/file_util.h"
+#include "base/path_service.h"
 #include "base/strings/string_util.h"
 #include "base/strings/stringprintf.h"
 #include "base/strings/utf_string_conversions.h"
 #include "base/threading/thread_restrictions.h"
+#include "chrome/common/chrome_paths.h"
+#include "chrome/common/logging_chrome.h"
 
 namespace base {
 
 typedef HMODULE (WINAPI* LoadLibraryFunction)(const wchar_t* file_name);
 
 namespace {
 
+PROC g_orig_create_process = NULL;
+HINSTANCE flash_dll;
+
+bool IsValidDosHeader(PIMAGE_DOS_HEADER dosHeader) {
+  if (IsBadReadPtr(dosHeader, sizeof(IMAGE_DOS_HEADER)))
+    return false;
+
+  if (dosHeader->e_magic != IMAGE_DOS_SIGNATURE)
+    return false;
+
+  return true;
+}
+
+bool IsValidNTHeader(PIMAGE_NT_HEADERS ntHeader) {
+  if (IsBadReadPtr(ntHeader, sizeof(PIMAGE_NT_HEADERS)))
+    return false;
+
+  if (ntHeader->Signature != IMAGE_NT_SIGNATURE)
+    return false;
+
+  return true;
+}
+
+BOOL WINAPI ReplaceFunctionAddressInModule(HMODULE module, PROC origFunc, PROC newFunc) {
```

Comment:
It failed to compile on some platforms (64bit, or nacl) due to warnings:
http://buildbot-master.node-webkit.org:8010/builders/nw21_win64/builds/32/steps/ninja_release/logs/stdio
http://buildbot-master.node-webkit.org:8010/builders/nw21_sdk_win32/builds/31/steps/ninja_release/logs/stdio

And please use base::win::IATPatchFunction rather than using own code for patching.